### PR TITLE
Add final payment and improve audit log performance

### PR DESCRIPTION
**Pull request by iqfx**
Most important things to note:
- Confirmation tokens are no longer valid when the payment has been paid. (downpayment or finalpayment)
- Payments have more metadata now.
- No separate view / route for final payment anymore.

Things to fix:
- Queuing emails no longer works because this application runs in docker. Docker image needs to be altered with supervisord support.

Things to do:
- Test (extensively)

### CODE REVIEW

**Review by BryanAafjes — APPROVED** [review]
Looks good to me!

File: app/Http/Controllers/ConfirmationController.php [hunk 1]
```diff
@@ -48,23 +61,24 @@ public function confirm(Request $request): Response|RedirectResponse
...
             }
 
-            $confirmationToken->confirmed = true;
-            $confirmationToken->save();
-            $this->participantController->store($request);
-            if(!$this->paymentController->checkIfParticipantPaid($user)) {
-                return $this->paymentController->payForReis($confirmationToken->id, Setting::where('name','Aanbetaling')->first()->value);
+            if(!$user->hasCompletedFinalPayment() && $user->hasCompletedDownPayment()) {
+                $this->participantController->store($request);
+                return $this->paymentController->payForReis($confirmationToken->id, Setting::where('name', 'Aanbetaling')->first()->value, PaymentTypes::FinalPayment());
```
Comment (azogaj) [inline app/Http/Controllers/ConfirmationController.php]: This needs to be the different setting!

File: app/Mail/parentMailSignup.php [hunk 2]
```diff
@@ -32,7 +32,7 @@ public function __construct(Participant $participant)
      */
     public function build()
     {
-        return $this->markdown('mails/parentSignup', ['participant' => $this->participant])->subject('Intro 2022 bevestiging')
+        return $this->markdown('mails/parentSignup', ['participant' => $this->participant])->subject('Reis bevestiging')
```
Comment (azogaj) [inline app/Mail/parentMailSignup.php]: English!

**Review by BryanAafjes — APPROVED** [review]
Looks good to me!

**Pull request merged**